CAT/model/IRT.py

```diff
@@ -8,13 +8,60 @@
 import numpy as np
 import math
 import torch.utils.data as data
+import torch.optim as optim
+import copy as cp
+import torch.nn.functional as F
 from math import exp as exp
 from sklearn.metrics import roc_auc_score
 from scipy import integrate
 import time
 from CAT.model.abstract_model import AbstractModel
 from CAT.dataset import AdapTestDataset, TrainDataset, Dataset
 from sklearn.metrics import accuracy_score
+from collections import namedtuple
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+SavedAction = namedtuple('SavedAction', ['log_prob', 'value'])
+
+def hard_sample(logits, dim=-1):
+    y_soft = F.softmax(logits, dim=-1)
+    index = y_soft.max(dim, keepdim=True)[1]
+    y_hard = torch.zeros_like(y_soft).scatter_(dim, index, 1.0)
+    ret = y_hard - y_soft.detach() + y_soft
+    return ret, index
+
+class Actor(nn.Module):
```

Comment:
Can these new strategies be applied to other CDMs such as NCD? If they can, I would recommend to add another file such as `utils.py` to write these related code instead of writing them in the file for IRT